fontir/src/glyph.rs

```diff
@@ -1688,6 +1730,57 @@ mod tests {
         assert_eq!(contour, &expected);
     }
 
+    #[test]
+    fn non_export_component_has_intermediate_layer() {
+        let _ = env_logger::builder().is_test(true).try_init();
+        // https://github.com/googlefonts/fontc/issues/1592
+        let [loc1, intermediate, loc2] = make_wght_locations([0.0, 0.5, 1.0]);
+
+        let mut aogonek = TestGlyph::new("Aogonek");
+        aogonek
+            .add_var_component(
+                "A.ogonekAccent",
+                &[(&loc1, Affine::IDENTITY), (&loc2, Affine::IDENTITY)],
+            )
+            .add_var_component(
+                "ogonekcomb.case",
+                &[(&loc1, Affine::IDENTITY), (&loc2, Affine::IDENTITY)],
+            );
+
+        let mut aogonek_accent = TestGlyph::new("A.ogonekAccent");
+        aogonek_accent
+            .add_var_contour(&[(&loc1, simple_square_path()), (&loc2, simple_square_path())])
+            .emit_to_binary(false);
+
+        let mut ogonekcomb_case = TestGlyph::new("ogonekcomb.case");
+        ogonekcomb_case.add_var_component(
+            "ogonek",
+            &[(&loc1, Affine::IDENTITY), (&loc2, Affine::IDENTITY)],
+        );
+        let mut ogonek = TestGlyph::new("ogonek");
+        ogonek.add_var_contour(&[
+            (&loc1, simple_square_path()),
+            (
+                &intermediate,
+                Affine::translate((0.0, 10.0)) * simple_square_path(),
+            ),
+            (
+                &loc2,
+                Affine::translate((0.0, 100.0)) * simple_square_path(),
+            ),
+        ]);
+
+        let context = test_context_with_locations(vec![loc1.clone(), loc2.clone()]);
+        context.glyphs.set(aogonek.0.clone());
+        context.glyphs.set(aogonek_accent.0);
+        context.glyphs.set(ogonekcomb_case.0);
+        context.glyphs.set(ogonek.0);
+
+        flatten_all_non_export_components(&context).unwrap();
```

Comment:
btw I think this function should really be renamed decompose_all_non_export_components because flatten is used in fontmake and elsewhere to mean making a nested component tree flat (with depth=1)